### PR TITLE
VM windows - update api versions and create bicep file

### PR DESCRIPTION
**Pull request by mumian**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@ju-shim - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@ju-shim - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

File: demos/vmss-windows-webapp-dsc-autoscale/main.bicep [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+@description('Admin username on all VMs.')
+param adminUsername string = 'vmssadmin'
+
+@description('Admin password on all VMs.')
+@secure()
+param adminPassword string
+
+@description('The base URI where artifacts required by this template are located. For example, if stored on a public GitHub repo, you\'d use the following URI: https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vmss-windows-webapp-dsc-autoscale/.')
+param artifactsLocation string = deployment().properties.templateLink.uri
```
Comment (bmoore-msft) [inline demos/vmss-windows-webapp-dsc-autoscale/main.bicep:35]: this needs to start with an underscore as in the original template

File: demos/vmss-windows-webapp-dsc-autoscale/main.bicep [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+@description('The sasToken required to access _artifactsLocation.  If your artifacts are stored on a public repo or public storage account you can leave this blank.')
+@secure()
+param artifactsLocationSasToken string = ''
```
Comment (bmoore-msft) [inline demos/vmss-windows-webapp-dsc-autoscale/main.bicep:39]: this needs to start with an underscore as in the original template

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@ju-shim - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by mumian** [comment]
@bmoore-msft - why can't I use the format()function?  The error is "Function'format' found within 'applicationUrl Line: 408, Column: 6".

**Comment by bmoore-msft** [comment]
because the property name is "url" it expects the uri function instead...

**Comment by azure-quickstarts** [comment]
@ju-shim - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@ju-shim - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request closed without merging**